pkg/controller/datavolume-conditions.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2020 The CDI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+const (
+	boundFalse      = "BoundChangeFalse"
+	transferRunning = "TransferRunning"
+	pvcBound        = "PVCBound"
+	pvcPending      = "PVCPending"
+	claimLost       = "ClaimLost"
+	notFound        = "NotFound"
+)
+
+func findConditionByType(conditionType cdiv1.DataVolumeConditionType, conditions []cdiv1.DataVolumeCondition) *cdiv1.DataVolumeCondition {
+	for i, condition := range conditions {
+		if condition.Type == conditionType {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+func updateCondition(conditions []cdiv1.DataVolumeCondition, conditionType cdiv1.DataVolumeConditionType, status corev1.ConditionStatus, message, reason string) []cdiv1.DataVolumeCondition {
+	condition := findConditionByType(conditionType, conditions)
+	if condition == nil {
+		conditions = append(conditions, cdiv1.DataVolumeCondition{
+			Type: conditionType,
+		})
+		condition = findConditionByType(conditionType, conditions)
+	}
+	if condition.Status != status {
+		condition.LastTransitionTime = metav1.Now()
+	}
+	condition.Status = status
+	condition.Message = message
+	condition.Reason = reason
+	return conditions
+}
+
+func updateRunningCondition(conditions []cdiv1.DataVolumeCondition, anno map[string]string) []cdiv1.DataVolumeCondition {
+	condition := findConditionByType(cdiv1.DataVolumeRunning, conditions)
+	if condition == nil {
+		conditions = append(conditions, cdiv1.DataVolumeCondition{
+			Type: cdiv1.DataVolumeRunning,
+		})
+		condition = findConditionByType(cdiv1.DataVolumeRunning, conditions)
+	}
+	if val, ok := anno[AnnRunningConditionMessage]; ok {
+		condition.Message = val
+	} else {
+		condition.Message = ""
+	}
```

Comment:
above if/else is equivalent to `val := anno[AnnRunningConditionMessage]`

Reply:
Hmm I thought it would error if the key didn't exist, but it just return blank. will fix.